pmpro-woocommerce.php

```diff
@@ -738,13 +738,36 @@ function pmprowoo_woocommerce_after_checkout_registration_form() {
  */
 function pmprowoo_update_user_meta( $meta_id, $object_id, $meta_key, $meta_value ) {
 	//tracks updates that are made
-	global $pmprowoo_updated_user_meta;
+	global $pmprowoo_updated_user_meta, $pmpro_pages, $pmpro_requirebilling;
 	if ( empty( $pmprowoo_updated_user_meta ) ) {
 		$pmprowoo_updated_user_meta = array();
 	}
 	if ( empty( $pmprowoo_updated_user_meta[ $object_id ] ) ) {
 		$pmprowoo_updated_user_meta[ $object_id ] = array();
 	}
+
+	/**
+	 * Check to see if customer is purchasing on the PMPro checkout page. Make sure that billing fields are required.
+	 * Check if custom fields are also present via RH or another solution and then allow updating of WooCommerce billing details.
+	 */
+	if ( ( $_REQUEST['level'] || is_page( $pmpro_pages['checkout'] ) ) && ! $pmpro_requirebilling ) {
```

Comment:
Is it possible that `$pmpro_pages` may not yet be set up or `$pmpro_pages['checkout']` may not be set at this point? Could potentially happen with custom scripts running early?

Reply:
```suggestion
	if (
		(
			! empty( $_REQUEST['level'] )
			|| (
				! empty( $pmpro_pages['checkout'] )
				&& is_page( $pmpro_pages['checkout'] )
			)
		)
		&& ! $pmpro_requirebilling
	) {
```

Reply:
Right, makes sense to check if the page isn't empty.